packages/battery/lib/battery.dart

```diff
@@ -0,0 +1,53 @@
+import 'dart:async';
+
+import 'package:flutter/services.dart';
+
+/// Indicates the current battery state.
+enum BatteryState { full, charging, discharging }
+
+class Battery {
+  factory Battery() {
+    if (_instance == null) {
+      MethodChannel methodChannel =
+          const MethodChannel('plugins.flutter.io/battery');
+      EventChannel eventChannel =
+          const EventChannel('plugins.flutter.io/charging');
+      _instance = new Battery.private(methodChannel, eventChannel);
+    }
+    return _instance;
+  }
+
+  Battery.private(this._methodChannel, this._eventChannel);
+
+  static Battery _instance;
+
+  final MethodChannel _methodChannel;
+  final EventChannel _eventChannel;
+  Stream<BatteryState> _onBatteryStateChanged;
+
+  /// Returns the current battery level in percent.
+  Future<int> get batteryLevel =>
+      _methodChannel.invokeMethod('getBatteryLevel');
+
+  /// Fires whenever the battery state changes.
+  Stream<BatteryState> get onBatteryStateChanged {
+    if (_onBatteryStateChanged == null) {
```

Comment:
Can't you just make the `Stream<BatteryState>` a `static final` field and rely on Dart's lazy initialization of such?

Basically my question is: do we need a `Battery` object at all?

Maybe we do, to be able to mock it when testing dependent libs. Otherwise those test would have to know about the channel names used in `Battery`.